### ISSUE TITLE
Test: Test Related Code

### ISSUE THREAD

**Issue by connor4312** [microsoft/vscode#223150]
ref https://github.com/microsoft/vscode/issues/126932

- [x] anyOS @aiday-mar 
- [x] anyOS @rzhao271

Complexity: 2

[Create Issue](https://github.com/microsoft/vscode/issues/new?body=Testing+%23223150%0A%0A&assignees=connor4312)

---

This iteration, I've added some API and UI to support relating code and its tests. I've adopted this for the selfhost test provider that we use in VS Code. Note that that the selfhost test provider has a basic, import-based implementation of related code. Please don't judge the relevancy of results it provides, just the editor functionality!

- Open VS Code and install the `@recommended` selfhost test provider if you haven't yet
- In some implementation code, right click and "Go to Related Test" / "Peek > Peek Related Test". Verify the UI displays well for you. 
- If you have code not imported by any tests anywhere, then you will get a cursor hint.
- In some test code, verify the opposite, that "Go to Related Code" / "Peek > Peek Related Code" works.
- In some implementation code, verify that **Test: Run Test at Cursor** runs the tests found via related code

**Comment by aiday-mar** [comment]
Hi @connor4312 I am trying to follow the steps of this TPI. I have installed the extension, I right click on code that is imported into a test, but I do not see under the Peek category, the `Peek Related Test` command, or the opposite `Peek Related Code` command. Here is a screen recording of what I see. Let me know if there is something I am missing,

https://github.com/user-attachments/assets/8b317ee2-37c5-4729-b83e-707de26a0088

**Comment by aiday-mar** [comment]
After discussion with colleagues, they suggested this could be happening because I was not on the main branch during testing. I switched to the main vscode branch and pulled main, then I reloaded the window. After doing this I saw the command appear in the context menu.

**Comment by cwebster-99** [comment]
Hey @connor4312, I am trying to test this and running into a couple of issues. 
I tried testing in Codespaces first but am not seeing any options after installing the extension and ensuring I am on the main branch per the comments above. 
![image](https://github.com/user-attachments/assets/4b34a151-eed8-4f69-981f-c152be863045)

I also tried to test this locally, but running into an issue with activating the extension 
![image](https://github.com/user-attachments/assets/84e9a550-e680-4b8c-8b3c-6bb05879db81)

Are there additional setup/config steps I need to take to get these options?
